extension/traverse-blame-dom.js

```diff
@@ -0,0 +1,18 @@
+/* globals utils */
+
+'use strict';
+
+window.traverseBlameDOM = (() => {
+	return $commitLink => {
+		const $lines = $commitLink.parent().parent().nextUntil('.blame-commit');
+		const $firstAndLast = [];
+		$firstAndLast.push($lines.eq(0));
```

Comment:
The whole thing does actually two things:

- traversing over the DOM in order to find all affected lines
- gathering the first and last affected line for the hash URL.

Especially when moving the whole functionality into its own file as suggested by @hkdobrev , private functions in the ``window.traverseBlameDOM`` scope should be created in order to segregate the steps.